lib/db/schemaupdater.go

```diff
@@ -629,5 +669,304 @@ func (db *schemaUpdater) updateSchemaTo12(_ int) error {
 		return err
 	}
 
+	if err := db.rewriteGlobals(t); err != nil {
+		return err
+	}
+
 	return t.Commit()
 }
+
+func (db *schemaUpdater) rewriteGlobals(t readWriteTransaction) error {
+	it, err := t.NewPrefixIterator([]byte{KeyTypeGlobal})
+	if err != nil {
+		return err
+	}
+	for it.Next() {
+		var vl VersionListDeprecated
+		if err := vl.Unmarshal(it.Value()); err != nil {
+			// If we crashed during an earlier migration, some version
+			// lists might already be in the new format: Skip those.
+			var nvl VersionList
+			if nerr := nvl.Unmarshal(it.Value()); nerr == nil {
+				continue
+			}
+			return err
+		}
+		if len(vl.Versions) == 0 {
+			if err := t.Delete(it.Key()); err != nil {
+				return err
+			}
+		}
+
+		var newVl VersionList
```

Comment:
This thing might do well as a `func convertVersionList(vl DeprecatedVersionList) VersionList` (with a readOnlyTransaction as well I guess as required)...?